src/client/src/components/chatpage.js

```diff
@@ -12,20 +12,24 @@ const USERNAMES_PLACEHOLDER = ''
 class ChatPage extends React.Component {
 	constructor(props) {
 		super(props)
+		const initProps = this.props.location.state
 		this.state = {
 			connected: false,
 			input: '',
 			messages: [],
 			selectedRoom: '',
-			username: this.props.location.state.username,
+			username: _.has(initProps, 'username') ? initProps['username'] : null,
```

Comment:
`_.get(initProps, 'username', null)`?

Reply:
Searches initialprops sent to the component for a username variable, if none is found, sets it to null, triggering redirect to login